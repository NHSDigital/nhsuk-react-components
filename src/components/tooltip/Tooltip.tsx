import React, { HTMLProps } from 'react';
import classNames from 'classnames';

interface TooltipProps extends HTMLProps<HTMLDivElement> {
  textProps?: HTMLProps<HTMLSpanElement>;
  tooltip: string;
}

const Tooltip: React.FC<TooltipProps> = ({
  className,
  tooltip,
  textProps,
  children,
  id,
  ...rest
}) => (
  <div
    className={classNames('nhsuk-tooltip', className)}
    id={id}
    aria-labelledby={id === undefined ? undefined : `${id}-tooltip-text`}
    {...rest}
  >
    {children}
    <span
      id={id === undefined ? undefined : `${id}-tooltip-text`}
      className={classNames('nhsuk-tooltip__text', textProps?.className)}
      {...textProps}
    >
      {tooltip}
    </span>
  </div>
);

Tooltip.defaultProps = {
  textProps: {},
};

export default Tooltip;
