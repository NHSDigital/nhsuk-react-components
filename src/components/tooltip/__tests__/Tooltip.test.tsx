import React from 'react';
import {render} from '@testing-library/react'
import Tooltip from '..';

describe('Tooltip', () => {
  it('Matches Snapshot', () => {
    const component = render(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    expect(component.container).toMatchSnapshot();
  });

  it('Passes through IDs', () => {
    const componentWithoutId = render(<Tooltip tooltip="Tooltip!">Hover Over Me</Tooltip>);
    const componentWithId = render(
      <Tooltip tooltip="Tooltip!" id="test-id">
        Hover Over Me
      </Tooltip>,
    );

    expect(componentWithoutId.container.querySelector('#test-id')).toBeNull();
    expect(componentWithoutId.container).toMatchSnapshot();

    expect(componentWithId.container.querySelector('#test-id-tooltip-text')).toBeTruthy();
    expect(componentWithId.container).toMatchSnapshot();

  })
});
