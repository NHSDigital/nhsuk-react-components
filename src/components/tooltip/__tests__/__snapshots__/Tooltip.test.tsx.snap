// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Matches Snapshot 1`] = `
<div>
  <div
    class="nhsuk-tooltip"
  >
    Hover Over Me
    <span
      class="nhsuk-tooltip__text"
    >
      Tooltip!
    </span>
  </div>
</div>
`;

exports[`Tooltip Passes through IDs 1`] = `
<div>
  <div
    class="nhsuk-tooltip"
  >
    Hover Over Me
    <span
      class="nhsuk-tooltip__text"
    >
      Tooltip!
    </span>
  </div>
</div>
`;

exports[`Tooltip Passes through IDs 2`] = `
<div>
  <div
    aria-labelledby="test-id-tooltip-text"
    class="nhsuk-tooltip"
    id="test-id"
  >
    Hover Over Me
    <span
      class="nhsuk-tooltip__text"
      id="test-id-tooltip-text"
    >
      Tooltip!
    </span>
  </div>
</div>
`;
